test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManagerSpec.scala

```diff
@@ -0,0 +1,208 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.actors
+
+import java.time.LocalDateTime
+import org.scalacheck.ScalacheckShapeless._
+import akka.actor.{ActorRefFactory, ActorSystem, PoisonPill, Props}
+import akka.testkit.{ImplicitSender, TestKit, TestProbe}
+import cats.data.EitherT
+import com.miguno.akka.testing.VirtualTime
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+import uk.gov.hmrc.cgtpropertydisposals.actors.FakeChildMaker.NewChildCreated
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryConnectorActor.RetryCallToTaxEnrolmentService
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.{MarkTaxEnrolmentRetryRequestAsSuccess, RetryTaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.models.Address.UkAddress
+import uk.gov.hmrc.cgtpropertydisposals.models.{Address, Error, TaxEnrolmentRequest, sample}
+import uk.gov.hmrc.cgtpropertydisposals.repositories.TaxEnrolmentRetryRepository
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService
+
+import scala.concurrent.Future
+
+class TaxEnrolmentRetryManagerSpec
+    extends TestKit(ActorSystem())
+    with ImplicitSender
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with MockFactory {
+
+  override def afterAll: Unit =
+    TestKit.shutdownActorSystem(system)
+
+  val time: VirtualTime = new VirtualTime()
+
+  val mockTaxEnrolmentService    = mock[TaxEnrolmentService]
+  val mockTaxEnrolmentRepository = mock[TaxEnrolmentRetryRepository]
+
+  implicit val localDateArb: Arbitrary[LocalDateTime] = Arbitrary.apply[LocalDateTime](Gen.const(LocalDateTime.now()))
+  val taxEnrolmentRequest                             = sample[TaxEnrolmentRequest]
+
+  "A Tax Enrolment Retry Manager actor" must {
+
+    "send a message to the mongo actor if tax enrolnment call was successful" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+
+      val taxEnrolmentRequest = TaxEnrolmentRequest(
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        2,
+        0,
+        249237542,
+        LocalDateTime.now()
+      )
+
+      app ! MarkTaxEnrolmentRetryRequestAsSuccess(taxEnrolmentRequest)
+
+      mongoProbe.receiveN(2)
```

Comment:
you can assert the messages the mongo probe receives as well over here to make sure they're the right thing

Reply:
I had a problem here. I was trying to assert over the message type but it didn't work. This probe receives two different types of messages and I am interested in the type of the second message it receives. However, the assert fails because it tries to assert on the first message it receives only. I used receiveN because I know this probe receives exactly two messages.